### PR TITLE
Add toggle to show_hide optional fields in add-on config

### PR DESCRIPTION
**Pull request by ludeeus**
## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:
image
image

## Checklist

- [x] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: hassio/src/addon-view/config/hassio-addon-config.ts [hunk 1]
```diff
@@ -94,6 +119,17 @@ class HassioAddonConfig extends LitElement {
             ? ""
             : html` <div class="errors">Invalid YAML</div> `}
         </div>
+        ${hasHiddenOptions
+          ? html`<ha-formfield
+              label="Show unused optional configuration options"
+            >
+              <ha-checkbox
```
Comment (bramkragten) [inline hassio/src/addon-view/config/hassio-addon-config.ts:126]: I think this should be toggle instead of a checkbox

Comment (ludeeus) [reply hassio/src/addon-view/config/hassio-addon-config.ts:126]: I don't agree, but ok.

Comment (bramkragten) [reply hassio/src/addon-view/config/hassio-addon-config.ts]: Haha why?

Comment (ludeeus) [reply hassio/src/addon-view/config/hassio-addon-config.ts]: The checkbox feels more natural to me, but I don't care enough to challenge you on it, has been switched to switch

**Review by bramkragten — APPROVED** [review]

**Pull request merged**